### ISSUE TITLE
Adding test data evaluation throughout the training process

### ISSUE THREAD

**Issue by Noam-Diamant** [jbloomAus/SAELens#446]
### Proposal 

I'm working with your library, and it seemed to me that it might be worth adding an evaluation on the test data during the training process, on the same set of metrics as the evaluation performed on the training set (L2 loss, L1 loss, L0 etc.). This is not currently in the code, and all evaluations are performed on the training data only. I think this could contribute to the understanding of SAE.

### Motivation

I want to see the SAE's performance on the test set as well, not just on the train.

### Pitch

Evaluate the metrics performed on the training set also on the test set, at each step of the training, just as is done on the training set.

### Additional context
the following code is from sae_lens/training/sae_trainer.py. You can add the relevant code here so that the metrics are also commented out in the test set, and I assume this entails changes to additional files related to training.

```
def fit(self) -> TrainingSAE:
        pbar = tqdm(total=self.cfg.total_training_tokens, desc="Training SAE")

        self.activations_store.set_norm_scaling_factor_if_needed()

        # Train loop
        while self.n_training_tokens < self.cfg.total_training_tokens:
            # Do a training step.
            layer_acts = self.activations_store.next_batch()[:, 0, :].to(
                self.sae.device
            )
            self.n_training_tokens += self.cfg.train_batch_size_tokens

            step_output = self._train_step(sae=self.sae, sae_in=layer_acts)

            if self.cfg.log_to_wandb:
                self._log_train_step(step_output)
                self._run_and_log_evals()

            self._checkpoint_if_needed()
            self.n_training_steps += 1
            self._update_pbar(step_output, pbar)

            ### If n_training_tokens > sae_group.cfg.training_tokens, then we should switch to fine-tuning (if we haven't already)
            self._begin_finetuning_if_needed()

        # fold the estimated norm scaling factor into the sae weights
        if self.activations_store.estimated_norm_scaling_factor is not None:
            self.sae.fold_activation_norm_scaling_factor(
                self.activations_store.estimated_norm_scaling_factor
            )
            self.activations_store.estimated_norm_scaling_factor = None

        # save final sae group to checkpoints folder
        self.save_checkpoint(
            trainer=self,
            checkpoint_name=f"final_{self.n_training_tokens}",
            wandb_aliases=["final_model"],
        )

        pbar.close()
        return self.sae
```

### Checklist

- [V ] I have checked that there is no similar [issue](https://github.com/jbloomAus/SAELens/issues) in the repo (**required**)

**Comment by janmenjayap** [comment]
Hello,
I would like to work on this issue.

My proposed implementation plan:

- Add an optional, independent evaluation data provider to  `SAETrainer` , while preserving the current training-provider fallback for backward compatibility.
- Allow  `LanguageModelSAETrainingRunner`  to construct evaluation data from an explicit held-out dataset or split.
- Reuse the existing evaluator and evaluation cadence rather than evaluating every training step.
- Ensure evaluation does not consume or modify the training iterator.
- Log held-out metrics under a distinct prefix.
- Add tests using deliberately different training and evaluation activations, including a regression test that evaluation leaves training progress unchanged.
- Document configuration and computational-cost considerations.

I will keep the initial implementation minimal and dependency-free. Does this scope and API direction sound appropriate before I begin?
